addOns/commonlib/src/main/java/org/zaproxy/addon/commonlib/ScanRulesUtils.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2021 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.commonlib;
+
+import java.util.regex.Pattern;
+import net.htmlparser.jericho.HTMLElementName;
+import net.htmlparser.jericho.OutputDocument;
+import net.htmlparser.jericho.Source;
+import net.htmlparser.jericho.StartTag;
+import org.parosproxy.paros.network.HttpMessage;
+
+public final class ScanRulesUtils {
```

Comment:
- Be good to provide some JavaDoc in the class and public methods (and add the version), so one does not need to check the code to know what's being specifically checked.
- The name should be more generic IMO, all code can use these.

Reply:
The name was my suggestion based in n current use but you are right it doesn’t have to suggest that it is limited to scan rules. How about `HttpMessageUtils`?

Reply:
What's the main purpose? Currently it seems to be used to resource identification, I'd go with something with those terms if that's the only purpose.

Reply:
`ResourceIdentificationUtils` ?

Reply:
Sounds good.